src/main.rs

```diff
@@ -782,6 +796,85 @@ fn start_vmm(cmd_arguments: ArgMatches) -> Result<Option<String>, Error> {
     r.map(|_| api_socket_path)
 }
 
+// This is a best-effort solution to the latency induced by the RCU
+// synchronization that happens in the kernel whenever the file
+// descriptor table is full and needs to be expanded. The initial table
+// has 64 entries on amd64.
+// Rust programs that use {File,EventFd}::try_clone() to share handles
+// are impacted by this issue.
+// This behavior is quite noticeable in the restore snapshot scenario,
+// the latency is a big chunk of the total time required to start
+// cloud-hypervisor and to restore the snapshot.
+// The kernel has an optimization in the expand table code path where it
+// doesn't call synchronize_rcu() if there is only one thread in the
+// process.
+// To address this issue we attempt to expand the descriptor table at
+// application start, when it has only one thread.
```

Comment:
Let's be more explicit about the behavior of the function from the comments, e.g. including the size of the file descriptor table we intend to expand to.

Reply:
Updated the comments and the commit message to better reflect what is going on.